src/Avalonia.Controls/SplitView.cs

```diff
@@ -0,0 +1,473 @@
+        /// Defines the <see cref="Content"/> property
+        /// </summary>
+        public static readonly StyledProperty<IControl> ContentProperty =
+            AvaloniaProperty.Register<SplitView, IControl>(nameof(Content));
+
+        /// <summary>
+        /// Defines the <see cref="CompactPaneLength"/> property
+        /// </summary>
+        public static readonly StyledProperty<double> CompactPaneLengthProperty =
+            AvaloniaProperty.Register<SplitView, double>(nameof(CompactPaneLength), defaultValue: 48);
+
+        /// <summary>
+        /// Defines the <see cref="DisplayMode"/> property
+        /// </summary>
+        public static readonly StyledProperty<SplitViewDisplayMode> DisplayModeProperty =
+            AvaloniaProperty.Register<SplitView, SplitViewDisplayMode>(nameof(DisplayMode), defaultValue: SplitViewDisplayMode.Overlay);
+
+        /// <summary>
+        /// Defines the <see cref="IsPaneOpen"/> property
+        /// </summary>
+        public static readonly DirectProperty<SplitView, bool> IsPaneOpenProperty =
+            AvaloniaProperty.RegisterDirect<SplitView, bool>(nameof(IsPaneOpen),
+                x => x.IsPaneOpen, (x, v) => x.IsPaneOpen = v);
+
+        /// <summary>
+        /// Defines the <see cref="OpenPaneLength"/> property
+        /// </summary>
+        public static readonly StyledProperty<double> OpenPaneLengthProperty =
+            AvaloniaProperty.Register<SplitView, double>(nameof(OpenPaneLength), defaultValue: 320);
+
+        /// <summary>
+        /// Defines the <see cref="PaneBackground"/> property
+        /// </summary>
+        public static readonly StyledProperty<IBrush> PaneBackgroundProperty =
+            AvaloniaProperty.Register<SplitView, IBrush>(nameof(PaneBackground));
+
+        /// <summary>
+        /// Defines the <see cref="PanePlacement"/> property
+        /// </summary>
+        public static readonly StyledProperty<SplitViewPanePlacement> PanePlacementProperty =
+            AvaloniaProperty.Register<SplitView, SplitViewPanePlacement>(nameof(PanePlacement));
+
+        /// <summary>
+        /// Defines the <see cref="Pane"/> property
+        /// </summary>
+        public static readonly StyledProperty<IControl> PaneProperty =
+            AvaloniaProperty.Register<SplitView, IControl>(nameof(Pane));
+
+        /// <summary>
+        /// Defines the <see cref="UseLightDismissOverlayMode"/> property
+        /// </summary>
+        public static readonly StyledProperty<bool> UseLightDismissOverlayModeProperty =
+            AvaloniaProperty.Register<SplitView, bool>(nameof(UseLightDismissOverlayMode));
+
+        /// <summary>
+        /// Defines the <see cref="TemplateSettings"/> property
+        /// </summary>
+        public static readonly StyledProperty<SplitViewTemplateSettings> TemplateSettingsProperty =
+            AvaloniaProperty.Register<SplitView, SplitViewTemplateSettings>(nameof(TemplateSettings));
+
+        private bool _isPaneOpen;
```

Comment:
these fields should be at top.

Reply:
I thought fields went between the AvaloniaProperties and constructors?

Reply:
yes your right,